jujugui/static/gui/src/app/app.js

```diff
@@ -12,58 +13,119 @@ const urls = jaaslib.urls;
-const SvgIcon = require('../components/svg-icon/svg-icon');
-const Terminal = require('../components/terminal/terminal');
-const UserMenu = require('../components/user-menu/user-menu');
-const USSOLoginLink = require('../components/usso-login-link/usso-login-link');
-const Zoom = require('../components/zoom/zoom');
+const AddedServicesList = require('./components/added-services-list/added-services-list');
+const Charmbrowser = require('./components/charmbrowser/charmbrowser');
+const DeploymentBar = require('./components/deployment-bar/deployment-bar');
+const DeploymentFlow = require('./components/deployment-flow/deployment-flow');
+const EnvSizeDisplay = require('./components/env-size-display/env-size-display');
+const ExpandingProgress = require('./components/expanding-progress/expanding-progress');
+const HeaderBreadcrumb = require('./components/header-breadcrumb/header-breadcrumb');
+const HeaderLogo = require('./components/header-logo/header-logo');
+const HeaderSearch = require('./components/header-search/header-search');
+const Help = require('./components/help/help');
+const Inspector = require('./components/inspector/inspector');
+const ISVProfile = require('./components/isv-profile/isv-profile');
+const Lightbox = require('./components/lightbox/lightbox');
+const LocalInspector = require('./components/local-inspector/local-inspector');
+const Login = require('./components/login/login');
+const Logout = require('./components/logout/logout');
+const MachineView = require('./components/machine-view/machine-view');
+const ModelActions = require('./components/model-actions/model-actions');
+const ModalGUISettings = require('./components/modal-gui-settings/modal-gui-settings');
+const ModalShortcuts = require('./components/modal-shortcuts/modal-shortcuts');
+const NotificationList = require('./components/notification-list/notification-list');
+const Panel = require('./components/panel/panel');
+const PostDeployment = require('./components/post-deployment/post-deployment');
+const Profile = require('./components/profile/profile');
+const Sharing = require('./components/sharing/sharing');
+const Status = require('./components/status/status');
+const SvgIcon = require('./components/svg-icon/svg-icon');
+const Terminal = require('./components/terminal/terminal');
+const UserMenu = require('./components/user-menu/user-menu');
+const USSOLoginLink = require('./components/usso-login-link/usso-login-link');
+const Zoom = require('./components/zoom/zoom');
 
 /**
-    A mixin for the JujuGUI class.
-    Stores all of the component renderer and cleanup methods.
+    A component to render the app.
 */
-const ComponentRenderersMixin = superclass => class extends superclass {
+class App extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      lightbox: false,
+      hoveredService: null,
+      settingsModalVisible: false,
+      sharingVisible: false,
+      shortcutsModalVisible: false
```

Comment:
State that was previously handled by checking what was in the DOM is now handled via React.